omod/src/main/java/org/openmrs/web/controller/PortletController.java

```diff
@@ -195,43 +206,67 @@ public ModelAndView handleRequest(HttpServletRequest request, HttpServletRespons
 					}
 					
 					if (Context.hasPrivilege(PrivilegeConstants.GET_OBS)) {
-						List<Obs> patientObs = Context.getObsService().getObservationsByPerson(p);
-						model.put("patientObs", patientObs);
+						Integer limit = getLimitParameter(request, as, "dashboard.observations.maximumNumberToShow");
+						
+						Person person = (Person) p;
+						List<Person> persons = Collections.singletonList(person);
+						
+						// Get most recent observations using limit parameter
+						List<Obs> paginatedObs = Context.getObsService().getObservations(persons, null, null, null, null,
```

Comment:
According to the user interface you shared, what do you do with the startIndex?

Reply:
Select the page size

Reply:
Is the page size the same as startIndex?

Reply:
Yes, based on the link you shared I renamed pageSize to that

Reply:
Where are you passing the two parameters in the getObservations api call?

Reply:
We are only passing the limit and not startIndex directly in that method call

Reply:
Which means that if you are on the last page, you still load all obs from the database to memory.

Reply:
@dkayiwa Yes that's right, added that